src/persist-client/src/lib.rs

```diff
@@ -453,6 +456,25 @@ impl PersistClient {
         Ok(reader)
     }
 
+    /// Creates and returns a [BatchFetcher] for the given shard id.
+    #[instrument(level = "debug", skip_all, fields(shard = %shard_id))]
+    pub fn create_batch_fetcher<K, V, T, D>(&self, shard_id: ShardId) -> BatchFetcher<K, V, T, D>
```

Comment:
```suggestion
    /// Creates and returns a [BatchFetcher] for the given shard id. This method can be misused, and requires
    /// that you ensure XXXX.
    #[instrument(level = "debug", skip_all, fields(shard = %shard_id))]
    pub fn unchecked_create_batch_fetcher<K, V, T, D>(&self, shard_id: ShardId) -> BatchFetcher<K, V, T, D>
```

I think its fine to add methods that increase perf at the cost of misuse, as long as they are appropriately named and documented